File: Sum of length - GFG/sum-of-length.java
//{ Driver Code Starts
// Initial Template for Java

// Initial Template for Java

import java.io.*;
import java.util.*;

// Position this line where user code will be pasted.

// Driver class
class Array {

    // Driver code
    public static void main(String[] args) throws IOException {
        // Taking input using buffered reader
        BufferedReader br =
            new BufferedReader(new InputStreamReader(System.in));

        int testcases = Integer.parseInt(br.readLine());
        // looping through all testcases
        while (testcases-- > 0) {
            int n = Integer.parseInt(br.readLine());
            //            String line = br.readLine();
            //            String[] a2 = line.trim().split("\\s+");
            //            int n =Integer.parseInt(a2[0]);
            //            int k =Integer.parseInt(a2[1]);
            String line1 = br.readLine();
            String[] a1 = line1.trim().split("\\s+");
            long a[] = new long[n];
            for (int i = 0; i < n; i++) {
                a[i] = Long.parseLong(a1[i]);
            }
            Solution ob = new Solution();
            // ArrayList<Long> ans=ob.smallestDifferenceTriplet(a,b,c,n);
            long ans = ob.sumoflength(a, n);
            System.out.println(ans);
        }
    }
}

// } Driver Code Ends


class Solution 
{
    
    long sumoflength(long arr[], int n) 
    {
        long result = 0; 
        HashSet<Long>set = new HashSet<Long>();
        
        int i = 0; int j = 0;
        
        while (i<arr.length)
        {
            while( (j<arr.length)&&(!set.contains(arr[j]) ) )
            {
                set.add(arr[j]);
                j += 1;
            }
            result = result + ( (j-i)*(j-i+1) )/2;
            set.remove(arr[i]);
            i += 1;
            
        }
        return result;
    }
}

File: Sum of length - GFG/sum-of-length.py
#User function Template for python3

class Solution:
    def sumoflength(self,arr, n):
     
        # For maintaining distinct elements.
        s = []
     
        # Initialize ending point and result
        j = 0
        ans = 0
     
        # Fix starting point
        for i in range(n):
             
            # Find ending point for current
            # subarray with distinct elements.
            while (j < n and (arr[j] not in s)):
                s.append(arr[j])
                j += 1
     
            # Calculating and adding all possible
            # length subarrays in arr[i..j]
            ans += ((j - i) * (j - i + 1)) // 2
     
            # Remove arr[i] as we pick new
            # starting point from next
            s.remove(arr[i])
     
        return ans
      


#{ 
 # Driver Code Starts
#Initial Template for Python 3



t=int(input())
for _ in range(0,t):
    n=int(input())
    a = list(map(int,input().split()))
    ob=Solution()
    ans=ob.sumoflength(a,n)
    print(ans)

# } Driver Code Ends